app/controllers/accounts_controller.rb

```diff
@@ -1,13 +1,14 @@
 class AccountsController < ApplicationController
   before_action :set_account, only: %i[show edit update]
-
+  before_action :check_admin_access, only: %i[index]
   def index
     @accounts = Account.all
   end
 
   def show; end
 
   def new
+    redirect_to account_path(current_user.account) unless current_user.account.nil?
```

Comment:
- Here should be used `if`, not unless
- Maybe `current_account_path`?

Reply:
current_account_path not working in header_html